main.py

```diff
@@ -12,6 +12,11 @@ def search(request: Request):
 
 @app.get('/search')
 def search_flights(fly_from_city: str = Query(...), to_city: str = Query(...), date: str = Query(...)):
+    try:
+        api.validate_city_name(fly_from_city)
```

Comment:
This is too complex. Instead of doing this, have the UI render a dropdown of valid city names. Pick 10 valid city names for the Tequila API, that are the most popular travel destinations.